pkg/cri/sbserver/base/cri_base.go

```diff
@@ -0,0 +1,118 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package base
+
+import (
+	"flag"
+	"fmt"
+	"path/filepath"
+
+	"k8s.io/klog/v2"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/log"
+	criconfig "github.com/containerd/containerd/pkg/cri/config"
+	"github.com/containerd/containerd/pkg/cri/constants"
+	"github.com/containerd/containerd/platforms"
+	"github.com/containerd/containerd/plugin"
+	imagespec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// CRIBase contains common dependencies for CRI's runtime, image, and podsandbox services.
+type CRIBase struct {
+	// Config contains all configurations.
+	Config criconfig.Config
+	// Client is an instance of the containerd client
+	Client *containerd.Client
+}
+
+func init() {
+	config := criconfig.DefaultConfig()
+
+	// Base plugin that other CRI services depend on.
+	plugin.Register(&plugin.Registration{
+		Type:   plugin.GRPCPlugin,
+		ID:     "cri",
+		Config: &config,
+		Requires: []plugin.Type{
+			plugin.EventPlugin,
+			plugin.ServicePlugin,
+			plugin.NRIApiPlugin,
```

Comment:
It seems NRIAPI is depended by cri-runtime-service only, shall we move the requirements to there?